File: 프로그래머스/Lv.1/86491. 최소직사각형/최소직사각형.cpp
#include <vector>
using namespace std;
int solution(vector<vector<int>> sizes) {
    int max_width = 0, max_height = 0;

    for (const auto& size : sizes) {
        int width = (size[0] > size[1]) ? size[0] : size[1];
        int height = (size[0] > size[1]) ? size[1] : size[0];

        if (width > max_width) {
            max_width = width;
        }
        if (height > max_height) {
            max_height = height;
        }
    }
    
    return max_width * max_height;
}

File: 프로그래머스/Lv.1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    # 각 명함의 가로와 세로 길이 중 더 큰 길이를 가로로 설정
    max_width = max_height = 0
    for size in sizes:
        max_width = max(max_width, max(size))
        max_height = max(max_height, min(size))
    # 가장 큰 가로와 세로 길이를 곱해서 지갑 크기를 계산
    return max_width * max_height